### PR TITLE
Gateio: Add function GetHistoricalFundingRates

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Based off branch #1339 

Adds function in gateio `GetHistoricalFundingRates`

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/gateio/gateio_wrapper.go [hunk 1]
```diff
@@ -2168,6 +2169,77 @@ func (g *Gateio) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) e
...
+	if r.IncludePredictedRate {
+		return nil, fundingrate.ErrIncludePredictedRateNotSupported
+	}
+
+	fPair, err := g.FormatExchangeCurrency(r.Pair, r.Asset)
+	if err != nil {
+		return nil, err
+	}
+
+	records, err := g.GetFutureFundingRates(ctx, r.PaymentCurrency.String(), fPair, 0 /*Defaults to returning maximum records*/)
```
Comment (gloriousCode) [inline exchanges/gateio/gateio_wrapper.go:2205]: I get 100 back with `0`, and 1000 back when setting it. Could be a case by case basis, but better to put in the limit

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for expanding things

File: exchanges/gateio/gateio_test.go [hunk 2]
```diff
@@ -3455,3 +3456,87 @@ func TestGetLatestFundingRates(t *testing.T) {
...
+
+	history, err := g.GetHistoricalFundingRates(context.Background(), &fundingrate.HistoricalRatesRequest{
+		Asset:           asset.Futures,
+		Pair:            currency.NewPair(currency.ENJ, currency.USDT),
+		PaymentCurrency: currency.USDT,
+		StartDate:       time.Now().Add(-time.Hour * 16),
+		EndDate:         time.Now(),
+	})
+	if !errors.Is(err, common.ErrStartAfterEnd) {
+		t.Fatalf("received: %v, expected: %v", err, common.ErrStartAfterEnd)
```
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/gateio/gateio_test.go:3527]: Fine, I guess I will have to then.

**Review by gloriousCode — APPROVED** [review]
tACK! TySB

**Review by thrasher- — APPROVED** [review]
tACK, looking good!
image

**Pull request merged**